tlsn/benches/README.md

```diff
@@ -0,0 +1,122 @@
+# TLSNotary bench utilities
+
+This crate provides utilities for benchmarking protocol performance under various network conditions and usage patterns.
+
+As the protocol is mostly IO bound, it's important to track how it performs in low bandwidth and/or high latency environments. To do this we set up temporary network namespaces and add virtual ethernet interfaces which we can control using the linux `tc` (Traffic Control) utility.
+
+## Setup
+
+To start we must create network namespaces for the prover and verifier, respectively.
+
+```sh
+ip netns add prover-ns
+ip netns add verifier-ns
+```
+
+Then we create a pair of virtual ethernet interfaces and add them to their respective namespaces.
+
+```sh
+ip link add prover-veth type veth peer name verifier-veth
+ip link set prover-veth netns prover-ns
+ip link set verifier-veth netns verifier-ns
+```
+
+If successful you should be able to see each interface in its namespace. For example, to see the prover interface:
+
+```sh
+ip netns exec prover-ns ip link
+```
+
+Then, activate each interface (bring it up).
+
+```sh
+ip netns exec prover-ns ip link set prover-veth up
+ip netns exec verifier-ns ip link set verifier-veth up
+```
+
+Next we'll assign IP addresses to each interface and set default routes:
+
+```sh
+ip netns exec prover-ns ip addr add 10.10.1.0/24 dev prover-veth
+ip netns exec prover-ns ip route add default via 10.10.1.0 dev prover-veth
+ip netns exec verifier-ns ip addr add 10.10.1.1/24 dev verifier-veth
+ip netns exec verifier-ns ip route add default via 10.10.1.1 dev verifier-veth
+```
+
+Verify that everything worked by pinging between them:
+
+```sh
+ip netns exec prover-ns ping 10.10.1.1
+```
+
+## Clean up
+
+For future reference, you can clean up this configuration as shown below.
+
+First, delete each interface in its' respective namespace:
+
+```sh
+ip netns exec prover-ns ip link delete prover-veth
+ip netns exec verifier-ns ip link delete verifier-veth
```

Comment:
This step did not work on my machine. Is it maybe because of the asymmetry of this command `ip link add prover-veth type veth peer name verifier-veth`?